nixos/modules/tasks/filesystems.nix

```diff
@@ -239,6 +239,8 @@ in
       '';
     };
 
+    boot.initrd.gptAutoRoot.enable = lib.mkEnableOption (lib.mdDoc "systemd-gpt-auto-generator for the root file system");
```

Comment:
Maybe we should put this in `boot.initrd.systemd` since this will only ever work with the systemd initrd? I would also prefer if this was called `gptAutoGenerator` because then it's more obvious where this functionality comes from (most importantly, that it's not an abstraction we've built ourselves).